Minor edits to Data Streams docs (#1732)
smartcontractkit/documentation: src/features/data-streams/common/gettingStarted.mdx

import { Aside, CodeSample, CopyText, ClickToZoom } from "@components"

> This guide shows you how to read data from a Data Stream and validate and store the answer onchain. This example uses a [Chainlink Automation Log Trigger](/chainlink-automation/guides/log-trigger) to check for events that require data. For this example, the log trigger comes from a simple emitter contract. Chainlink Automation then uses `StreamsLookup` to retrieve a signed report from the Data Streams Engine, return the data in a callback, and run the [`performUpkeep` function](/chainlink-automation/reference/automation-interfaces#performupkeep-function-for-log-triggers) on your registered upkeep contract. The `performUpkeep` function calls the `verify` function on the verifier contract.

Reviewer: ```suggestion
This guide shows you how to read data from a Data Stream, validate the answer, and store the answer onchain. This example uses a [Chainlink Automation Log Trigger](/chainlink-automation/guides/log-trigger) to check for events that require data. For this example, the log trigger comes from a simple emitter contract. Chainlink Automation then uses `StreamsLookup` to retrieve a signed report from the Data Streams Engine, return the data in a callback, and run the [`performUpkeep` function](/chainlink-automation/reference/automation-interfaces#performupkeep-function-for-log-triggers) on your registered upkeep contract. The `performUpkeep` function calls the `verify` function on the verifier contract.
```

     style="max-width: 70%;"
   />

1. In the **Contract** section, select the `StreamsUpkeep` contract and fill in the **verifier proxy address** corresponding to the stream you want to read from. You can find this address on the [Stream IDs](/data-streams/stream-ids) page. The verifier proxy address for the ETH/USD stream on Arbitrum Sepolia is <CopyText text="0x2ff010DEbC1297f19579B4246cad07bd24F2488A" code/>.

   <ClickToZoom
     src="/images/data-streams/getting-started/deploy-upkeep.webp"
     alt="Chainlink Data Streams Remix Deploy Upkeep Contract"
     style="max-width: 70%;"
   />

1. Click the **Deploy** button to deploy the contract. MetaMask prompts you to confirm the transaction. Check the transaction details to ensure you deploy the contract to _Arbitrum Sepolia_.

1. After you confirm the transaction, the contract address appears under the **Deployed Contracts** list in Remix. Save this contract address for later.

   <ClickToZoom
     src="/images/data-streams/getting-started/deployed-upkeep-1.webp"
     alt="Chainlink Data Streams Remix Deployed Upkeep Contract"
     style="max-width: 70%;"
   />

## Deploy the emitter contract

> This contract emits logs that trigger the upkeep. This code can be part of your dApp. For example, you might emit log triggers when your users initiate a trade or other action requiring data retrieval. For this Getting Started guide, we will use a very simple emitter so you can test the upkeep and data retrieval.

Reviewer: ```suggestion
This contract emits logs that trigger the upkeep. This code can be part of your dApp. For example, you might emit log triggers when your users initiate a trade or other action requiring data retrieval. For this Getting Started guide, use a very simple emitter so you can test the upkeep and data retrieval.
```

Avoid "we" and just use imperative sentences for simplicity.

1. [Open the LogEmitter.sol](https://remix.ethereum.org/#url=https://docs.chain.link/samples/DataStreams/LogEmitter.sol) contract in Remix.

   <CodeSample src="samples/DataStreams/LogEmitter.sol" showButtonOnly />

1. Under the **Solidity Compiler** tab, select the `0.8.19` Solidity compiler and click the **Compile LogEmitter.sol** button to compile the contract.

   <ClickToZoom
     src="/images/data-streams/getting-started/compile-logemitter.webp"
     alt="Chainlink Data Streams Remix Compile Log Emitter Contract"
     style="max-width: 70%;"
   />

1. Open MetaMask and make sure the network is still set to _Arbitrum Sepolia_.

> 1. In Remix, under the **Deploy & Run Transactions** tab, ensure the **Environment** is still set to _Injected Provider - MetaMask_.

Reviewer: ```suggestion
1. On the **Deploy & Run Transactions** tab in Remix, ensure the **Environment** is still set to _Injected Provider - MetaMask_.
```

This one is just a style choice, but I think it simplifies the sentence.

## Fund the upkeep contract

In this example, the upkeep contract pays for onchain verification of reports from Data Streams. The Automation subscription does not cover the cost.

Open MetaMask and send 1 testnet LINK on _Arbitrum Sepolia_ to the upkeep contract address you saved earlier.

    <ClickToZoom
      src="/images/data-streams/getting-started/fund-deployed-upkeep.webp"
      alt="Chainlink Data Streams Fund Deployed Upkeep"
      style="max-width: 70%;"
    />

## Emit a log

Now, you can use your emitter contract to emit a log and initiate the upkeep, which retrieves data for the specified Data Streams asset ID.

1. In Remix, on the **Deploy & Run Transactions** tab, expand your emitter contract under the **Deployed Contracts** section.

   <ClickToZoom
     src="/images/data-streams/getting-started/emitter-emitlog.webp"
     alt="Chainlink Data Streams Emit Log"
     style="max-width: 70%;"
   />

1. Click the `emitLog` button to call the function and emit a log. MetaMask prompts you to accept the transaction.

> After the transaction is complete, the log is emitted, and the upkeep is triggered. You can find the upkeep transaction hash at [Chainlink Automation UI](https://automation.chain.link/arbitrum-sepolia). Check to make sure the transaction is successful.

Reviewer: ```suggestion
After the transaction is complete, the log is emitted, and the upkeep is triggered. You can find the upkeep transaction hash in the [Chainlink Automation UI](https://automation.chain.link/arbitrum-sepolia). Check to make sure the transaction is successful.
```

    <ClickToZoom src="/images/data-streams/getting-started/request-fulfilled.webp" />

## View the retrieved price

The retrieved price is stored as a variable in the contract and emitted in the logs.

> 1. In Remix, under the **Deploy & Run Transactions** tab, expand your upkeep contract in the **Deployed Contracts** section.

Reviewer: ```suggestion
1. On the **Deploy & Run Transactions** tab in Remix, expand the details of your upkeep contract in the **Deployed Contracts** section.
```

Style preference

1. Click the `last_retrieved_price` getter function to view the retrieved price. The answer on the ETH/USD feed uses 18 decimal places, so an answer of `248412100000000000` indicates an ETH/USD price of 2484.121. Each feed uses a different number of decimal places for answers. See the [Stream IDs](/data-streams/stream-ids) page for more information.

   <ClickToZoom
     src="/images/data-streams/getting-started/deployed-upkeep-2.webp"
     alt="Chainlink Data Streams Deployed Upkeep"
     style="max-width: 70%;"
   />

> Alternatively, you can view the price emitted in the logs for your upkeep transaction. You can find the upkeep transaction hash at [Chainlink Automation UI](https://automation.chain.link/arbitrum-sepolia) and view the logs in the [Arbitrum Sepolia explorer](https://sepolia.arbiscan.io/).

Reviewer: ```suggestion
Alternatively, you can view the price emitted in the logs for your upkeep transaction. You can find the upkeep transaction hash at [Chainlink Automation UI](https://automation.chain.link/arbitrum-sepolia) and view the transaction logs in the [Arbitrum Sepolia explorer](https://sepolia.arbiscan.io/).
```

## Examine the code

The example code you deployed has all the interfaces and functions required to work with Chainlink Automation as an upkeep contract. It follows a similar flow to the trading flow in the [Architecture](/data-streams#example-trading-flow) documentation but uses a basic log emitter to simulate the client contract that would initiate a `StreamsLookup`. After the contract receives and verifies the report, `performUpkeep` stores the price from the report in the `last_retrieved_price` and emits a `PriceUpdate` log message with the price. You could modify this to use the data in a way that works for your specific use case and application.

> The code example uses `revert` with `StreamsLookup` to convey call information about what streams to retrieve. See the [eip-3668 rationale](https://eips.ethereum.org/EIPS/eip-3668#rationale) for more information on `revert` use in this way.

Reviewer: ```suggestion
The code example uses `revert` with `StreamsLookup` to convey call information about what streams to retrieve. See the [EIP-3668 rationale](https://eips.ethereum.org/EIPS/eip-3668#rationale) for more information about how to use `revert` in this way.
```
